### PR TITLE
Update contributing guide with new label policy

### PR DESCRIPTION
**Pull request by apopiak**
+ Update contributing guide with the new label policy including `C` and `M` labels.
+ Update PR template with more concrete instructions on labels.

### CODE REVIEW

File: docs/CONTRIBUTING.adoc [hunk 1]
```diff
@@ -28,7 +28,8 @@ Merging pull requests once CI is successful:
...
 . PRs must be tagged with respect to _release notes_ with https://github.com/paritytech/substrate/labels/B0-silent[`B0-silent`] and `B1-..`. The former indicates that no changes should be mentioned in any release notes. The latter indicates that the changes should be reported in the corresponding release note
-. PRs that break the external API must be tagged with https://github.com/paritytech/substrate/labels/B2-breaksapi[`B2-breaksapi`], when it changes the FRAME or consensus of running system with https://github.com/paritytech/substrate/labels/B3-breaksconsensus[`B3-breaksconsensus`]
+. PRs that break the external API must be tagged with https://github.com/paritytech/substrate/labels/B2-breaksapi[`B2-breaksapi`], when it changes the FRAME or consensus of running system with https://github.com/paritytech/substrate/labels/B3-breaksconsensus[`B3-breaksconsensus`].
```
Comment (kianenigma) [inline docs/CONTRIBUTING.adoc:31]: I would rephrase this entirely similar to what Gav said, since i think it is way easier to remember: Label all PRs with one `A`, one `B` and one `C` at the very least.

File: docs/CONTRIBUTING.adoc [hunk 2]
```diff
@@ -28,7 +28,8 @@ Merging pull requests once CI is successful:
...
+. PRs should be labeled with their release importance via the `C1-C9` labels as well as their project/scope via the `M*` labels.
```
Comment (kianenigma) [inline docs/CONTRIBUTING.adoc:32]:
```suggestion
. PRs should be labeled with their release importance via the `C1-C9` labels as well as their project/scope via the `M*` labels.
```

Are we sure about the M labels? I ranted a lot about how they are confusing lately. Perhaps what we mean here is the [project](https://github.com/paritytech/substrate/projects) in github?

Comment (apopiak) [reply docs/CONTRIBUTING.adoc:32]: oooh, no idea :shrug: @gavofyork ?

Comment (gavofyork) [reply docs/CONTRIBUTING.adoc:32]: `M` labels are dead.

**Comment by gavofyork** [comment]
Sorry for the cross post :)

# PRs
- `A-*` Pull request status. @s3krit: Automation should ensure this is kept in sync with GH's native review system. ONE REQUIRED.
- `B-*` Changelog and/or Runtime-upgrade post composition markers. Automation uses this. ONE REQUIRED.
- `C-*` Release notes release-priority markers. Automation uses this. EXACTLY ONE REQUIRED.
- `D-*` More general tags on the PR denoting various implications and requirements.

# Issues
- `I-*` Issue severity and type. EXACTLY ONE REQUIRED.
- `P-*` Issue priority. AT MOST ONE ALLOWED.
- `Q-*` Issue difficulty. AT MOST ONE ALLOWED.
- `Z-*` More general tags on the issue, denoting context and resolution.

I have removed the `M-*` class as they have in any case been superseded by GH's Project functionality.

**Comment by apopiak** [comment]
While updating the PR template I noticed that it is actually a checklist, not a template.
Do we want to add something more template-like to the top? Suggestions?

File: docs/CONTRIBUTING.adoc (hunk 1 above)
Comment (apopiak) [reply docs/CONTRIBUTING.adoc:42]: Restructured the section, let me know what you think :-)
